### PR TITLE
MINOR: [Doc][Python] Update compute kernel list

### PR DESCRIPTION
**Pull request by jimexist**

### CODE REVIEW

File: docs/source/python/api/compute.rst [hunk 1]
```diff
@@ -191,6 +188,7 @@ characters of a given class.
    ascii_is_lower
    ascii_is_printable
    ascii_is_space
+   ascii_is_title
```
Comment (lidavidm) [inline docs/source/python/api/compute.rst:191]: Note the *_is_title functions are listed below.

**Review by lidavidm — COMMENTED** [review]
Thanks for adding some missing functions. Is there a reason why some functions are removed?

**Comment by Jimexist** [comment]
> Thanks for adding some missing functions. Is there a reason why some functions are removed?

they are reported missing while `make html` builds. and i can't find them in the python module.

**Comment by lidavidm** [comment]
Most of these functions are not explicitly written out, but are instead autogenerated at runtime. So they will not exist in the source code. What is the output from `make html`?

**Comment by Jimexist** [comment]
```
In [5]: set(dir(compute))
Out[5]:
{'ArraySortOptions',
 'CastOptions',
 'DayOfWeekOptions',
 'DictionaryEncodeOptions',
 'ElementWiseAggregateOptions',
 'ExtractRegexOptions',
 'FilterOptions',
 'Function',
 'FunctionOptions',
 'FunctionRegistry',
 'HashAggregateFunction',
 'HashAggregateKernel',
 'IndexOptions',
 'JoinOptions',
 'Kernel',
 'MakeStructOptions',
 'MatchSubstringOptions',
 'ModeOptions',
 'PadOptions',
 'PartitionNthOptions',
 'QuantileOptions',
 'ReplaceSliceOptions',
 'ReplaceSubstringOptions',
 'ScalarAggregateFunction',
 'ScalarAggregateKernel',
 'ScalarAggregateOptions',
 'ScalarFunction',
 'ScalarKernel',
 'SetLookupOptions',
 'SliceOptions',
 'SortOptions',
 'SplitOptions',
 'SplitPatternOptions',
 'StrptimeOptions',
 'TDigestOptions',
 'TakeOptions',
 'TrimOptions',
 'VarianceOptions',
 'VectorFunction',
 'VectorKernel',
 '__builtins__',
 '__cached__',
 '__doc__',
 '__file__',
 '__loader__',
 '__name__',
 '__package__',
 '__spec__',
 '_decorate_compute_function',
 '_get_arg_names',
 '_get_options_class',
 '_handle_options',
 '_make_generic_wrapper',
 '_make_global_functions',
 '_make_signature',
 '_wrap_function',
 'abs',
 'abs_checked',
 'acos',
 'acos_checked',
 'add',
 'add_checked',
 'all',
 'and',
 'and_',
 'and_kleene',
 'and_not',
 'and_not_kleene',
 'any',
 'array_filter',
 'array_sort_indices',
 'array_take',
 'ascii_center',
 'ascii_is_alnum',
 'ascii_is_alpha',
 'ascii_is_decimal',
 'ascii_is_lower',
 'ascii_is_printable',
 'ascii_is_space',
 'ascii_is_title',
 'ascii_is_upper',
 'ascii_lower',
 'ascii_lpad',
 'ascii_ltrim',
 'ascii_ltrim_whitespace',
 'ascii_reverse',
 'ascii_rpad',
 'ascii_rtrim',
 'ascii_rtrim_whitespace',
 'ascii_split_whitespace',
 'ascii_trim',
 'ascii_trim_whitespace',
 'ascii_upper',
 'asin',
 'asin_checked',
 'atan',
 'atan2',
 'binary_join',
 'binary_join_element_wise',
 'binary_length',
 'binary_replace_slice',
 'bit_wise_and',
 'bit_wise_not',
 'bit_wise_or',
 'bit_wise_xor',
 'call_function',
 'case_when',
 'cast',
 'ceil',
 'coalesce',
 'cos',
 'cos_checked',
 'count',
 'count_substring',
 'count_substring_regex',
 'day',
 'day_of_week',
 'day_of_year',
 'dedent',
 'dictionary_encode',
 'divide',
 'divide_checked',
 'ends_with',
 'equal',
 'extract_regex',
 'fill_null',
 'filter',
 'find_substring',
 'find_substring_regex',
 'floor',
 'function_registry',
 'get_function',
 'greater',
 'greater_equal',
 'hash_count',
 'hash_min_max',
 'hash_sum',
 'hour',
 'if_else',
 'index',
 'index_in',
 'index_in_meta_binary',
 'inspect',
 'invert',
 'is_finite',
 'is_in',
 'is_in_meta_binary',
 'is_inf',
 'is_nan',
 'is_null',
 'is_valid',
 'iso_calendar',
 'iso_week',
 'iso_year',
 'less',
 'less_equal',
 'list_flatten',
 'list_functions',
 'list_parent_indices',
 'list_value_length',
 'ln',
 'ln_checked',
 'log10',
 'log10_checked',
 'log1p',
 'log1p_checked',
 'log2',
 'log2_checked',
 'make_struct',
 'match_like',
 'match_substring',
 'match_substring_regex',
 'max_element_wise',
 'mean',
 'microsecond',
 'millisecond',
 'min_element_wise',
 'min_max',
 'minute',
 'mode',
 'month',
 'multiply',
 'multiply_checked',
 'nanosecond',
 'negate',
 'negate_checked',
 'not_equal',
 'or',
 'or_',
 'or_kleene',
 'pa',
 'partition_nth_indices',
 'power',
 'power_checked',
 'quantile',
 'quarter',
 'replace_substring',
 'replace_substring_regex',
 'replace_with_mask',
 'second',
 'shift_left',
 'shift_left_checked',
 'shift_right',
 'shift_right_checked',
 'sign',
 'sin',
 'sin_checked',
 'sort_indices',
 'split_pattern',
 'split_pattern_regex',
 'starts_with',
 'stddev',
 'string_is_ascii',
 'strptime',
 'subsecond',
 'subtract',
 'subtract_checked',
 'sum',
 'take',
 'tan',
 'tan_checked',
 'tdigest',
 'trunc',
 'unique',
 'utf8_center',
 'utf8_is_alnum',
 'utf8_is_alpha',
 'utf8_is_decimal',
 'utf8_is_digit',
 'utf8_is_lower',
 'utf8_is_numeric',
 'utf8_is_printable',
 'utf8_is_space',
 'utf8_is_title',
 'utf8_is_upper',
 'utf8_length',
 'utf8_lower',
 'utf8_lpad',
 'utf8_ltrim',
 'utf8_ltrim_whitespace',
 'utf8_replace_slice',
 'utf8_reverse',
 'utf8_rpad',
 'utf8_rtrim',
 'utf8_rtrim_whitespace',
 'utf8_slice_codeunits',
 'utf8_split_whitespace',
 'utf8_trim',
 'utf8_trim_whitespace',
 'utf8_upper',
 'value_counts',
 'variance',
 'warnings',
 'xor',
 'year'}
```

**Comment by lidavidm** [comment]
The docs are built against the corresponding source in cpp/, not against the most recent release. So for instance `product` isn't there since it was added after 5.0.0.

**Comment by lidavidm** [comment]
There is not a guide for building everything including the docs. But you will have to at least build Arrow/C++ and PyArrow from source: https://arrow.apache.org/docs/developers/python.html#building-on-linux-and-macos

File: docs/source/python/api/compute.rst [hunk 2]
```diff
@@ -105,8 +104,6 @@ variants which detect domain errors.
    log1p_checked
    log2
    log2_checked
-   logb
-   logb_checked
```
Comment (lidavidm) [inline docs/source/python/api/compute.rst:109]: These kernels definitely exist in the development branch. We should not remove them from the list.

File: docs/source/python/api/compute.rst (hunk 1 above)
Comment (lidavidm) [reply docs/source/python/api/compute.rst:191]: Follow up here - these are already listed in the docs below.

**Review by lidavidm — CHANGES_REQUESTED** [review]
Thanks for going through the list. As I mentioned, you will not see some of these kernels in 5.0.0 because the docs are written for the next release. You will want to build the docs against a development build of PyArrow. There are instructions in the docs: https://arrow.apache.org/docs/developers/python.html#building-on-linux-and-macos

**Pull request closed without merging**